src/tools/lambert.rs

```diff
@@ -40,10 +43,87 @@ pub struct LambertSolution {
     pub phi: f64,
 }
 
-/// Solves the Lambert boundary problem using a standard secant method.
+/// Calculate the direction multiplier based on the transfer kind.
+///
+/// # Arguments
+///
+/// * `kind` - The kind of transfer (auto, short way, long way, or number of revolutions).
+/// * `r_final` - The final radius vector.
+/// * `r_init` - The initial radius vector.
+///
+/// # Returns
+///
+/// `Result<f64, NyxError>` - The direction multiplier or an error if the transfer kind is not supported.
+fn calculate_dm(
+    kind: &TransferKind,
+    r_final: &Vector3<f64>,
+    r_init: &Vector3<f64>,
+) -> Result<f64, NyxError> {
+    match kind {
+        TransferKind::Auto => {
+            let mut dnu = r_final[1].atan2(r_final[0]) - r_init[1].atan2(r_final[1]);
+            if dnu > TAU {
+                dnu -= TAU;
+            } else if dnu < 0.0 {
+                dnu += TAU;
+            }
+
+            if dnu > std::f64::consts::PI {
+                Ok(-1.0)
+            } else {
+                Ok(1.0)
+            }
+        }
+        TransferKind::ShortWay => Ok(1.0),
+        TransferKind::LongWay => Ok(-1.0),
+        _ => Err(NyxError::LambertMultiRevNotSupported),
+    }
+}
+
+/// Calculate the new values of phi, c2, and c3 based on the current value of phi.
+///
+/// # Arguments
+///
+/// `phi` - The current value of phi.
+///
+/// # Returns
+///
+/// `(f64, f64, f64)` - The new values of phi, c2, and c3.
+fn calculate_phi_and_c2_c3(phi: f64) -> (f64, f64, f64) {
```

Comment:
I don't think this should be its own function: part of the Secant method is updating these values, and someone familiar with the secant method would expect to find this at the bottom of the loop.

Reply:
The function is still in the code I think, although the functionality has been restored to the `standard` function.

Reply:
Oups - done.